src-backend/api/signals/page_pre_save.py

```diff
@@ -0,0 +1,16 @@
+from django.dispatch import receiver
+from django.db.models.signals import pre_save
+from django.db.models import F
+from api.models import Page
+import logging
+
+logger = logging.getLogger('debugger')
+
+
+@receiver(pre_save, sender=Page)
+def on_page_pre_save(sender, instance=None, raw=False, **kwargs):
+    # Update index of all objects with display index >= object we are saving
+    if instance.display_index is not None:
```

Comment:
So are we sending back every attribute in a `PUT` everytime? If that's the case, then I can add extra checks to see if that attribute is actually being modified or not.